common/app/common/templates/TemplateService.java

```diff
@@ -0,0 +1,28 @@
+package common.templates;
+
+import common.pages.PageData;
+import play.twirl.api.Html;
+
+/**
+ * Service that provides HTML pages, using some kind of template system.
+ */
+public interface TemplateService {
```

Comment:
That was the initial name but I realized it was wrong. This service only knows about templates. So you give it a template name and data to inject, and returns the HTML. It doesn't actually know about views. Who knows about views is, for example, the `ProductCatalogView`, which knows the view you want to obtain, so it calls the proper template and requests the proper data and delivers the HTML.